src/WebJobs.Script.WebHost/Configuration/ExtensionSystemOptions.cs

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Azure.WebJobs.Hosting;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Configuration
+{
+    /// <summary>
+    /// Represents the system options for an individual extension.
+    /// </summary>
+    public sealed class ExtensionSystemOptions : IOptionsFormatter
+    {
+        private static readonly JsonSerializerOptions Options = CreateJsonOptions();
+        private AuthorizationLevel _webhookAuthorizationLevel = AuthorizationLevel.System;
+
+        /// <summary>
+        /// Gets or sets the name of the extension these options apply to.
+        /// </summary>
+        public string ExtensionName { get; set; }
```

Comment:
I added the name here, primarily to ensure that when we log the options the name is shown, which wouldn't have been the case after moving to the new options snapshot pattern.